src/Kdyby/Doctrine/DI/OrmExtension.php

```diff
@@ -762,18 +772,12 @@ protected function getServiceBoundManagers(Nette\DI\ServiceDefinition $def)
 
 	public function afterCompile(Code\ClassType $class)
 	{
-		$init = $class->methods['initialize'];
-
-		$init->addBody('Kdyby\Doctrine\Diagnostics\Panel::registerBluescreen($this);');
+		if ($this->isTracyPresent()) {
+			$init = $class->methods['initialize'];
+			$init->addBody('Kdyby\Doctrine\Diagnostics\Panel::registerBluescreen($this);');
 
-		if (property_exists('Tracy\BlueScreen', 'collapsePaths')) {
-			$blueScreen = 'Tracy\Debugger::getBlueScreen()';
-			$commonDirname = dirname(Nette\Reflection\ClassType::from('Doctrine\Common\Version')->getFileName());
-
-			$init->addBody($blueScreen . '->collapsePaths[] = ?;', array(dirname(Nette\Reflection\ClassType::from('Kdyby\Doctrine\Exception')->getFileName())));
-			$init->addBody($blueScreen . '->collapsePaths[] = ?;', array(dirname(dirname(dirname(dirname($commonDirname)))))); // this should be vendor/doctrine
-			foreach ($this->proxyAutoloaders as $dir) {
-				$init->addBody($blueScreen . '->collapsePaths[] = ?;', array($dir));
+			if (property_exists('Tracy\BlueScreen', 'collapsePaths')) {
```

Comment:
I see no point in checking if `BlueScreen` is present, if `IBarPanel` is already checked.